### PR TITLE
Cleanup RectangleSelector example.

### PR DESCRIPTION
**Pull request by anntzer**
- Don't use "q" or "a" as keys given that they are already mapped to
  "quit" and "all_axes" (even though the latter is deprecated).
- Document the interactivity directly as the axes title.
- No need to plot too many things in the axes; they're irrelevant.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/widgets/rectangle_selector.py [hunk 1]
```diff
@@ -30,23 +30,23 @@ def line_select_callback(eclick, erelease):
...
 N = 100000  # If N is large one can see improvement by using blitting.
 x = np.linspace(0.0, 10.0, N)
 
 ax.plot(x, +np.sin(.2*np.pi*x), lw=3.5, c='b', alpha=.7)  # plot something
-ax.plot(x, +np.cos(.2*np.pi*x), lw=3.5, c='r', alpha=.5)
-ax.plot(x, -np.sin(.2*np.pi*x), lw=3.5, c='g', alpha=.3)
-
-print("\n      click  -->  release")
+ax.set_title(
+    "Click and release to draw a rectangle.\n"
```
Comment (timhoffm) [inline examples/widgets/rectangle_selector.py:48]:
```suggestion
    "Click and drag to draw a rectangle.\n"
```

**Comment by anntzer** [comment]
the bug is tracked at https://github.com/matplotlib/matplotlib/issues/17476.

File: examples/widgets/rectangle_selector.py (hunk 1 above)
Comment (anntzer) [reply examples/widgets/rectangle_selector.py:48]: yup

**Comment by QuLogic** [comment]
The example in the class docstring for `RectangleSelector` uses these same keys, and should probably be synced with this example.

**Comment by timhoffm** [comment]
Possibly, the example in the class docstring should be replaced by a link to this example.

**Comment by anntzer** [comment]
indeed, just made a link instead.

**Pull request merged**